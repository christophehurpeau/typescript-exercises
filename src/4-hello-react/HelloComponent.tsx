import React, { useState } from 'react';
import InputName from './InputName';

export default function HelloComponent() {
  const [name, setName] = useState('World');
  return (
    <div>
      <span>Hello {name}!</span>
      <InputName value={name} onChange={(e) => setName(e.target.value)} />
    </div>
  );
}
