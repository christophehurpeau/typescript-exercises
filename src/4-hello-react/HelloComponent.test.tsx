/*
 * @jest-environment jsdom
 */

import React from 'react';
import renderer, { act } from 'react-test-renderer';
import HelloComponent from './HelloComponent';

describe('HelloComponent', () => {
  it('should match snapshot', () => {
    const component = renderer.create(<HelloComponent />);

    expect(component.toJSON()).toMatchSnapshot();

    const componentInstance = component.root;

    const input = componentInstance.findByType('input');

    act(() => {
      input.props.onChange({ target: { value: 'John' } });
    });

    expect(component.toJSON()).toMatchSnapshot();
  });
});
