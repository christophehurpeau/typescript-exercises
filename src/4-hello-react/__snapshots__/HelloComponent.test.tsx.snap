// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloComponent should match snapshot 1`] = `
<div>
  <span>
    Hello 
    World
    !
  </span>
  <input
    autoComplete="off"
    onChange={[Function]}
    onKeyUp={[Function]}
    type="text"
    value="World"
  />
</div>
`;

exports[`HelloComponent should match snapshot 2`] = `
<div>
  <span>
    Hello 
    John
    !
  </span>
  <input
    autoComplete="off"
    onChange={[Function]}
    onKeyUp={[Function]}
    type="text"
    value="John"
  />
</div>
`;
